tap_universal_file/streams.py

```diff
@@ -0,0 +1,463 @@
+
+            yield (
+                self.ModifiedDictReader(
+                    f=self._skip_rows(file_name),
+                    delimiter=delimiter,
+                    quotechar=quote_character,
+                    fieldnames=override_headers,
+                    config=self.config,
+                ),
+                file_name,
+                file["last_modified"],
+            )
+
+    def _skip_rows(self, file: str) -> list[str]:
+        """Takes a file name and only returns rows which are not configured as skipped.
+
+        Args:
+            file: The name of the file to process.
+
+        Returns:
+            A list containing rows from the file.
+        """
+        with self.fs_manager.filesystem.open(
+            path=file,
+            mode="rt",
+            compression=self.get_compression(file=file),
+        ) as f:
+            file_list = []
+            file_list.extend(f)
+        try:
+            for _ in range(self.config["delimited_header_skip"]):
+                file_list.pop(0)
+            for _ in range(self.config["delimited_footer_skip"]):
+                file_list.pop()
+        except IndexError:
+            return []
+        else:
+            return file_list
+
+    class ModifiedDictReader(csv.DictReader):
+        """A modified version of DictReader that detects improperly formatted rows."""
+
+        def __init__(  # noqa: PLR0913
+            self,
+            f: Any,  # noqa: ANN401
+            fieldnames: Any | None = None,
+            restkey: Any | None = None,
+            restval: Any | None = None,
+            dialect: str = "excel",
+            config: dict = None,
+            *args: Any,
+            **kwds: Any,
+        ) -> None:
+            """Identical to the superclass's method except for defining self.config."""
+            super().__init__(f, fieldnames, restkey, restval, dialect, *args, **kwds)
+            self.config = config if config is not None else {}
+
+        def __next__(self) -> dict:
+            """Identical to the superclass's method except for raising FileFormatErrors.
+
+            Raises:
+                FileFormatError: If a row in the *SV has too few entries.
+                FileFormatError: If a row in the *SV has too many entries.
+
+            Returns:
+                A dictionary containing the records for a row.
+            """
+            if self.line_num == 0:
+                self.fieldnames  # Used for its side-effect. # noqa: B018
+            row = next(self.reader)
+            self.line_num = self.reader.line_num
+            while row == []:
+                row = next(self.reader)
+            d = dict(zip(self.fieldnames, row))
+            lf = len(self.fieldnames)
+            lr = len(row)
+            if lf < lr:
+                if self.config["delimited_error_handling"] == "fail":
+                    msg = (
+                        f"Too few entries at line {self.line_num}. To suppress this "
```

Comment:
Having the Filename here would be very helpful as well. 

This may be a case where instead of `RuntimeError`'s we may want a custom exception (Or the `FileFormatError` may work? But I don't know what else the `super` uses that Exception for) , and you could catch this [here](https://github.com/MeltanoLabs/tap-universal-file/pull/20/files#diff-893f897bd0df36e495cd9d0443825598759089135ab5eaba77c777f2d9bf5004R96) and also add the `file_name` to the message (along with the path) to help folks debug this. Not sure how needed or not this is

Reply:
Hit this while trying the tap out, and I can verify you really really want to know which `file` it's talking about and it's not easy to tell

Reply:
Kept it as RuntimeError but made sure to increase the debug information to an appropriately detailed level.